sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -80,6 +80,14 @@ public static ColumnVector allocate(int capacity, DataType type, MemoryMode mode
     public int length;
     public int offset;
 
+    // reused buffer to return a primitive array
```

Comment:
I'm not sure if we need this buffer. `UnsafeArrayData.toXXXArray` doesn't use a buffer, and it seems safer to always return a new array.